### PR TITLE
{math}[intel/2015a] mathematical libraries + deps (REVIEW)

### PR DESCRIPTION
**Pull request by gjbex**
- Armadillo
- arpack-ng
- gettext
- libffi
- GLib
- Mesa
- libXt
- libSM
- libICE
- libXdamage
- libXfixes
- fixesproto
- libdrm
- libpciaccess
- xorg-macros
- glproto
- libxml2
- makedepend
- xrandr
- libXrandr
- randrproto
- libXrender
- renderproto
- libXext
- libXdmcp
-  CGAL

### CODE REVIEW

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb [hunk 1]
```diff
@@ -1,19 +1,32 @@
+#easyblock = 'CMakeMake'
```
Comment (boegel) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: please remove

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
+
 name = 'Qt'
 version = '4.8.6'
 
 homepage = 'http://qt-project.org/'
 description = "Qt is a comprehensive cross-platform C++ application framework."
 
-toolchain = {'name': 'intel', 'version': '2015a'}
+toolchain = {'name': 'intel', 'version': '2015a'}
```
Comment (boegel) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: trailing whitespace

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb [hunk 3]
```diff
@@ -1,19 +1,32 @@
...
 description = "Qt is a comprehensive cross-platform C++ application framework."
 
-toolchain = {'name': 'intel', 'version': '2015a'}
+toolchain = {'name': 'intel', 'version': '2015a'} 
 
-source_urls = [
-    'http://origin.releases.qt-project.org/qt4/source/',
-    'http://master.qt-project.org/archive/qt/%(version_major_minor)s/%(version)s/'
-]
+source_urls = ['http://download.qt-project.org/official_releases/qt/4.8/4.8.6/']
```
Comment (boegel) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: use `%(version)s` rather than hardcoding the version

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb [hunk 4]
```diff
@@ -1,19 +1,32 @@
...
 sources = ['%(namelower)s-everywhere-opensource-src-%(version)s.tar.gz']
 
-patches = ['Qt-4.8.6_icc-qUnused.patch']
 
-dependencies = [('GLib', '2.40.0')]
+dependencies = [
+    ('GLib', '2.42.1'),
+    ('Mesa', '7.11.2', '-Python-2.7.9'),
```
Comment (boegel) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: this Mesa version is quite old... I have an *almost* working newer version of Mesa in #1226, but I got stuck since it requires libudev, and I haven't figured out yet how to flesh that out of the systemd source tarball it's now shipped in (yes, ...)

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb [hunk 5]
```diff
@@ -1,19 +1,32 @@
...
+    ('xrandr', '1.4.3'),
+    ('libXrender','0.9.8'),
+]
+
+patches = [ 'Qt-4.8.6_icc-qUnused.patch' ]
+# patches = [ 'Qt-4.8.5-foss-2014a-OpenGL-qmake.patch2']
+
+configopts =" -confirm-license -opensource -silent -opengl -glib -webkit -developer-build -xrender"
```
Comment (boegel) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: this significantly extends the Qt functionality; that's good, but it also pulls in a bunch of other dependencies...

especially Mesa is quite painful, so I'm not sure whether including it as a dependendy for Qt by default is a good idea

maybe this should be a variant with a custom versionsuffix, rather than a 'standard' build?

thoughts, @gjbex?

**Comment by boegel** [comment]
@gjbex: besides the concerns with Qt/Mesa, this is going to be difficult to get merged in, because it includes so a big stack of easyconfigs... We should look into breaking this up in several PRs which are easier to handle. A max of somewhere in between 5-10 easyconfigs should be used as a rule of thumb, with 'difficult' ones like Mesa or an enhanced version of Qt getting a dedicated PR.

Let me know what you think about this (either in this issue or via IM).

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb (hunk 1 above)
Comment (gjbex) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: Ok, but why?  That seems somewhat inconsistent with having to add easyblock = 'ConfigureMake'.

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb (hunk 4 above)
Comment (gjbex) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: The _almost_ is exactly why I've not moved to a higher version ;)

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb (hunk 5 above)
Comment (gjbex) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: A number of software packages rely on a Qt with OpenGL support, hence this choice.  Webkit would be optional, but than again, it doesn't really matter anymore.

**Comment by gjbex** [comment]
Unfortunately I got lost in ever extending circles of dependency hell, all caused by the Qt dependency of GCAL.  (In fact, it made me read the Wikipedia article on Dante's Inferno, and I'm not joking here).  If this is broken up, we'll end up with quite a number of PRs that depend on one another.  I don't know whether that is to be preferred.

File: easybuild/easyconfigs/g/GLib/GLib-2.42.1-intel-2015a.eb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+version = '2.42.1'
+
+homepage = 'http://www.gtk.org/'
+description = """GLib is one of the base libraries of the GTK+ project"""
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+toolchainopts = {'optarch': True, 'pic': True}
+
+source_urls = ['http://ftp.gnome.org/pub/gnome/sources/glib/%(version_major_minor)s/']
+sources = ['glib-%(version)s.tar.xz']
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/GLib/GLib-2.42.1-intel-2015a.eb]: `SOURCELOWER_TAR_XZ` ?

File: easybuild/easyconfigs/l/libICE/libICE-1.0.9-intel-2015a.eb [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+dependencies = [
+    ('xtrans', '1.3.5'),
+]
+
+builddependencies = [
+    ('xproto', '7.0.27'),
+]
+
+
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libICE/libICE-1.0.9-intel-2015a.eb]: drop newline

File: easybuild/easyconfigs/c/CGAL/CGAL-4.5.1-intel-2015a-Python-2.7.9.eb [hunk 8]
```diff
@@ -0,0 +1,30 @@
+name = 'CGAL'
+version = '4.5.1'
+
+homepage = 'http://www.cgal.org/'
+description = """The goal of the CGAL Open Source Project is to provide easy access to efficient 
+ and reliable geometric algorithms in the form of a C++ library."""
+
+
```
Comment (wpoely86) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.5.1-intel-2015a-Python-2.7.9.eb]: drop newline

File: easybuild/easyconfigs/c/CGAL/CGAL-4.5.1-intel-2015a-Python-2.7.9.eb [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+toolchainopts = {'strict': True}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://fenicsproject.org/pub/software/contrib/']
+
+pythonversion = '2.7.9'
+versionsuffix = "-Python-%s" % pythonversion
+
+dependencies = [
+    ('GMP', '5.1.3'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.5.1-intel-2015a-Python-2.7.9.eb]: why not version 6?

File: easybuild/easyconfigs/l/libXext/libXext-1.3.3-intel-2015a.eb [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+]
+
+builddependencies = [
+    ('xproto','7.0.27'),
+    ('xextproto', '7.3.0'),
+]
+
+sanity_check_paths = {
+    'files': ['include/X11/extensions/%s' % x for x in [
+            'dpms.h', 'extutil.h', 'MITMisc.h', 'multibuf.h', 'security.h', 'shape.h', 'sync.h', 'Xag.h', 'Xcup.h',
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libXext/libXext-1.3.3-intel-2015a.eb]: make sure the lines doesn't go over column 120

File: easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+version = '1.4.2'
+
+description = "libXrandr - X Resize, Rotate and Reflection extension library"
+homepage = 'http://cgit.freedesktop.org/xorg/lib/libXrandr/'
+
+# The following are automatically calculated - cross fingers
+sources = ['libXrandr-1.4.2.tar.gz']
+source_urls = ['http://xorg.freedesktop.org/archive/individual/lib/','download']
+
+toolchain = {'name': 'intel', 'version': '2015a'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb]: toolchain below description. Try to do:
- homepage
- description
- toolchain

File: easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+sanity_check_paths = {
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb]: put this below the deps

File: easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    ('libXext', '1.3.3'),   
+    ('libXrender', '0.9.8'),
+    ('libX11', '1.6.2', '-Python-2.7.9'),
+    ('randrproto', '1.4.3'),
+    ('renderproto', '0.11.1')
+]
+           
+
+parallel = 1 # this is a conservative choice
+maxparallel = 1 # this is a conservative choice
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb]: why both?

File: easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+
+moduleclass = 'vis'
+
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb]: delete empty line

File: easybuild/easyconfigs/l/libXt/libXt-1.1.4-intel-2015a.eb [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+    ('xproto', '7.0.27'),
+    ('kbproto', '1.0.6'),
+]
+
+sanity_check_paths = {
+    'files' : ['include/X11/%s' % x for x in [
+        'CallbackI.h', 'CompositeP.h', 'Constraint.h', 'Core.h', 'CreateI.h', 'HookObjI.h', 'Intrinsic.h',
+        'IntrinsicP.h', 'ObjectP.h', 'RectObj.h', 'ResConfigP.h', 'SelectionI.h', 'ShellI.h', 'StringDefs.h',
+        'TranslateI.h', 'Vendor.h', 'Xtos.h', 'Composite.h', 'ConstrainP.h', 'ConvertI.h', 'CoreP.h', 'EventI.h',
+        'InitialI.h', 'IntrinsicI.h', 'Object.h', 'PassivGraI.h', 'RectObjP.h', 'ResourceI.h', 'Shell.h', 'ShellP.h',
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libXt/libXt-1.1.4-intel-2015a.eb]: watch the 120 chars limit on a line

File: easybuild/easyconfigs/l/libdrm/libdrm-2.4.59-intel-2015a.eb [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+
+source_urls  = ['http://dri.freedesktop.org/libdrm/']
+sources = [SOURCELOWER_TAR_GZ]
+
+dependencies = [
+    ('libpthread-stubs', '0.3'),
+    ('libpciaccess', '0.13.2'),
+]
+
+configopts = '--enable-nouveau-experimental-api'
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libdrm/libdrm-2.4.59-intel-2015a.eb]: why is this needed?

File: easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-intel-2015a-Python-2.7.9.eb [hunk 17]
```diff
@@ -0,0 +1,57 @@
+easyblock = 'ConfigureMake'
+
+name = 'Mesa'
+version = '7.11.2'
+
+homepage = 'http://www.mesa3d.org/'
+description = """Mesa is an open-source implementation of the OpenGL specification -
+ a system for rendering interactive 3D graphics."""
+
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-intel-2015a-Python-2.7.9.eb]: drop a empty line

File: easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-intel-2015a-Python-2.7.9.eb [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+    ('xproto', '7.0.27'),
+    ('libXdamage', '1.1.4'),
+    ('libXfixes', '5.0.1'),
+]
+
+configopts = " --disable-osmesa --enable-glu --disable-gallium-llvm --disable-gallium-gbm --enable-glx --with-driver=xlib"
+configopts += " --disable-driglx-direct --with-gallium-drivers='' --without-demos --disable-egl"""
+
+# package-config files for os dependencies are in an os specific place
+preconfigopts = ' PKG_CONFIG_PATH="$PKG_CONFIG_PATH:/usr/lib64/pkgconfig/:/usr/share/pkgconfig" '
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-intel-2015a-Python-2.7.9.eb]: yes, but doesn't pkg-config look already in those places?

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb [hunk 19]
```diff
@@ -6,14 +6,25 @@ description = "Qt is a comprehensive cross-platform C++ application framework."
...
+
+sanity_check_paths = {
+    'files': ['lib/libQtCore.so'],
+	 'dirs': [],
```
Comment (wpoely86) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: indenting

File: easybuild/easyconfigs/r/randrproto/randrproto-1.4.3-intel-2015a.eb [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['.'] # fixme when the compilation is successful
+}
+
+parallel = 1 # this is a conservative choice
+maxparallel = 1 # this is a conservative choice
+
+moduleclass = 'vis'
+
```
Comment (wpoely86) [inline easybuild/easyconfigs/r/randrproto/randrproto-1.4.3-intel-2015a.eb]: drop empty line

File: easybuild/easyconfigs/r/randrproto/randrproto-1.4.3-intel-2015a.eb [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+    ('libX11', '1.6.2', '-Python-2.7.9')
+]
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['.'] # fixme when the compilation is successful
+}
+
+parallel = 1 # this is a conservative choice
+maxparallel = 1 # this is a conservative choice
```
Comment (wpoely86) [inline easybuild/easyconfigs/r/randrproto/randrproto-1.4.3-intel-2015a.eb]: both needed?

File: easybuild/easyconfigs/x/xrandr/xrandr-1.4.3-intel-2015a.eb [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+    ('libXrandr', '1.4.2')
+]
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['.'] # fixme when the compilation is successful
+}
+
+parallel = 1 # this is a conservative choice
+maxparallel = 1 # this is a conservative choice
```
Comment (wpoely86) [inline easybuild/easyconfigs/x/xrandr/xrandr-1.4.3-intel-2015a.eb]: both needed?

File: easybuild/easyconfigs/x/xrandr/xrandr-1.4.3-intel-2015a.eb [hunk 23]
```diff
@@ -0,0 +1,29 @@
...
+name = 'xrandr'
+version = '1.4.3'
+
+description = "xrandr-1.3.5: Primitive command line interface to RandR extension"
+homepage = 'http://xorg.freedesktop.org/'
+
+sources = ['xrandr-1.3.5.tar.bz2']
+source_urls = ['http://pkgs.fedoraproject.org/repo/pkgs/xorg-x11-server-utils/xrandr-1.3.5.tar.bz2/9735173a84dca9b05e06fd4686196b07/']
+
+toolchain = {'name': 'intel', 'version': '2015a'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/x/xrandr/xrandr-1.4.3-intel-2015a.eb]: toolchain below description

File: easybuild/easyconfigs/c/CGAL/CGAL-4.5.1-intel-2015a-Python-2.7.9.eb (hunk 9 above)
Comment (gjbex) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.5.1-intel-2015a-Python-2.7.9.eb]: All other dependencies are on this version of GMP.  There were issues with the Intel build, which have been resolved after deciding to go with 5.1.3 to avoid trouble.

File: easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb (hunk 13 above)
Comment (gjbex) [inline easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb]: Ask the original author ;)

File: easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-intel-2015a-Python-2.7.9.eb (hunk 18 above)
Comment (gjbex) [inline easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-intel-2015a-Python-2.7.9.eb]: Someone must have had a reason to add it. Given that nothing should break if it is there, leave it as is?

Comment (wpoely86) [inline easybuild/easyconfigs/m/Mesa/Mesa-7.11.2-intel-2015a-Python-2.7.9.eb]:
```
PKG_CONFIG_PATH

    A colon-separated (on Windows, semicolon-separated) list of directories to search for .pc files. The default 
directory will always be searched after searching the path; the default is libdir/pkgconfig:datadir/pkgconfig where 
libdir is the libdir where pkg-config and datadir is the datadir where pkg-config was installed. 
```

The trouble with these things is that we end up with a bunch of cruft and nobody really knows why it's there (or even if it's still needed).

File: easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb (hunk 13 above)
Comment (wpoely86) [inline easybuild/easyconfigs/l/libXrandr/libXrandr-1.4.2-intel-2015a.eb]: I would drop the `maxparallel`. That really doesn't have any use if `parallel` is already set.

**Comment by boegel** [comment]
@gjbex: `easyblock = 'ConfigureMake'` is missing in `libXdamage-1.1.4-intel-2015a.eb`

File: easybuild/easyconfigs/x/xrandr/xrandr-1.4.3-intel-2015a.eb (hunk 22 above)
Comment (boegel) [inline easybuild/easyconfigs/x/xrandr/xrandr-1.4.3-intel-2015a.eb]: only one makes sense, I'd prefer `parallel = 1`, since `maxparallel` is intended for situations where parallel building *does* work, but only up to a certain level

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb (hunk 4 above)
Comment (boegel) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: So, should we keep this on hold until the newer Mesa is ready?

File: easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb (hunk 5 above)
Comment (boegel) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.6-intel-2015a.eb]: Well, should we then make this a custom `Qt` version, with `versionsuffix = '-opengl'`, and use that for stuff that requires OpenGL-enabled Qt?

Dragging in `Mesa` into standard Qt is just too much imho...

**Comment by boegel** [comment]
@gjbex: dependency hell, I've been there... I understand that you'd rather not spend more time on this, to avoid going deeper into Dante's Inferno, but this is going to be really hard to merge in like this, which will likely result in it rotting away rather than ending up being merged.

It seems like the `Armadillo` and `arpack-ng` easyconfigs included in here have nothing to do with the others, so those can easily be fleshed out in a dedicated PR?

Also, how about splitting off the Mesa easyconfig + all the X11-related deps into a dedicated config too? That would effectively cut this in half, making it more manageable.

That would suffice imho...

You can leave this PR as is, and just update it as the Armadillo/arpak-ng and Mesa PRs are merged into develop, no need to split of the remaining easyconfigs into yet another PR.

**Comment by migueldiascosta** [comment]
@gjbex, this PR is being closed for the following reason(s): no activity for > 6 months.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer
